js/ui/windowManager.js

```diff
@@ -2205,6 +2223,33 @@ var WindowManager = new Lang.Class({
             this._hideOtherWindows(actor, animateFade);
     },
 
+    _clipActor : function(origin, position, actor) {
+        switch (origin) {
```

Comment:
I'd condense this switch statement by pre-setting `clipX, clipY, clipWidth, clipHeight` to the default values, and just update the values of the affected variables according to the switch case. Meaning we'd only call `actor.set_clip(...)` once.

Reply:
I've updated this switch to modify the `clipX, clipY, clipWidth, clipHeight` and do only one `actor.set_clip`.